client.go

```diff
@@ -51,6 +51,14 @@ import (
 // ClientParam specifies functional argument for client
 type ClientParam func(c *Client) error
 
+// Tracer sets request tracer
```

Comment:
`Tracer sets a request tracer constructor`

Seems like a complication unless it is aiming to use the client in concurrent environment. It is only meaningful if the rest of Client is also safe for concurrent use - is this so?